dali/python/nvidia/dali/pipeline.py

```diff
@@ -247,6 +256,19 @@ def _prepare_graph(self):
         for output in outputs:
             Edge._validate_edge_reference(output)
 
+        unique_outputs = []
+        output_map = {}
+        next_free_idx = 0
+        for name in outputs:
+            idx = output_map.get(name, next_free_idx + 1)
+            if idx > next_free_idx:
+                output_map[name] = next_free_idx
+                idx = next_free_idx
+                next_free_idx += 1
+                unique_outputs.append(name)
+            self._outputs_list.append(idx)
```

Comment:
I wanted to suggest something that is clearer when indexing, but it does lookup the dictionary twice.
```
for name in outputs:
    if name not in output_map:
        output_map[name] = next_free_idx
        next_free_idx +=1
        unique_outputs.append(name)
     self._outputs_list.append(output_map[name])
```

Reply:
It is not perf critical. Done.